### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by etosch**
Non-backwards compatible Amidar tests + new intervention API.

### CODE REVIEW

**Comment by etosch** [comment]
Currently it's the Breakout tests that are failing. Blocked by https://github.com/toybox-rs/toybox-rs/issues/3, since the Game interfaces expects a minimum set of keys exported.

File: ctoybox/toybox/toybox/envs/atari/#base.py# [hunk 1]
```diff
@@ -0,0 +1,152 @@
+from abc import ABC, abstractmethod
```
Comment (jjfiv) [inline ctoybox/toybox/toybox/envs/atari/#base.py#]: Delete these files: #*#

File: ctoybox/toybox/toybox/interventions/core.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    self.direction = direction
+
+  def decode(intervention, direction, clz):
+    return Direction(intervention, direction)
+
+  def encode(self):
+    return self.direction
+
+
+class Vec2D(BaseMixin):
```
Comment (jjfiv) [inline ctoybox/toybox/toybox/interventions/core.py]: This is part of Breakout, not core, I thought...

**Comment by etosch** [comment]
BTW this build is now failing because of an issue on the rust side -- when we load up a game, it now starts with zero balls.

File: ctoybox/toybox/toybox/interventions/core.py (hunk 2 above)
Comment (etosch) [reply ctoybox/toybox/toybox/interventions/core.py]: Is it?

File: ctoybox/toybox/toybox/interventions/base.py [hunk 3]
```diff
@@ -1,41 +1,121 @@
...
+from abc import ABC, abstractmethod
+import inspect
 from ctoybox import Toybox
-
 import json
 """ Contains the base class for interventions. 
 
 To make interventions for a new game, subclass Intervention."""
 
+class BaseMixin(ABC):
```
[the remark this answers is not in the document]
Comment (etosch) [reply ctoybox/toybox/toybox/interventions/base.py]: Motion to make that part of another PR?

File: .gitignore [hunk 4]
```diff
@@ -31,3 +31,8 @@ toybox-regress-models*.zip
...
...921082615.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.2959785456.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.2959785456.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3283232304.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3283232304.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3294615773.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3294615773.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3485671521.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3485671521.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3771075072.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.3771075072.2019-05-18.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.845090117.2019-05-12.model toybox/toybox/sample_tests/models/AmidarToyboxNoFrameskip-v4.ppo2.5e7.845090117.2019-05-18.model
```
[the remark this answers is not in the document]
Comment (jjfiv) [reply .gitignore]: I committed a fix from the web gui.

File: ctoybox/toybox/toybox/envs/atari/base.py [hunk 5]
```diff
@@ -109,6 +110,11 @@ def step(self, action_index):
         assert(type(self._action_set)== list)
     
         self.toybox.apply_ale_action(self._action_set[action_index])
+
+        if self.ale.game_over():
+            print('GAME OVER')
```
[the remark this answers is not in the document]
Comment (jjfiv) [reply ctoybox/toybox/toybox/envs/atari/base.py]: #153.

File: ctoybox/toybox/toybox/interventions/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjfiv) [reply ctoybox/toybox/toybox/interventions/base.py]: #152.

File: ctoybox/toybox/toybox/interventions/core.py (hunk 2 above)
Comment (jjfiv) [reply ctoybox/toybox/toybox/interventions/core.py]: It's a candidate to move to core, we just haven't had a reason.

**Pull request merged**